tools/tileset_test.py

```diff
@@ -0,0 +1,365 @@
+                f'"{entry["typeid"]}" ]'
+            if key != len(self.furniture) - 1:
+                retstr += ','
+            retstr += '\n'
+        retstr += identstr + '\t],\n'
+
+        retstr += identstr * ident + '\t"traps": [\n'
+        for key, entry in enumerate(self.traps):
+            retstr += identstr
+            retstr += "\t\t[ " + \
+                f'{entry["row"]}, {entry["column"]}, ' \
+                f'"{entry["typeid"]}" ]'
+            if key != len(self.traps) - 1:
+                retstr += ','
+            retstr += '\n'
+        retstr += identstr + '\t]\n'
+
+        retstr += identstr + '}'
+        return retstr
+
+    def __str__(self) -> str:
+        return self.as_string()
+
+
+class FullMap:
+
+    def __init__(self, offset_x: int, offset_y: int):
+        self._map = []
+        self._map.append(QuarterMap(offset_x, offset_y))
+        self._map.append(QuarterMap(offset_x, offset_y + 1))
+        self._map.append(QuarterMap(offset_x + 1, offset_y))
+        self._map.append(QuarterMap(offset_x + 1, offset_y + 1))
+
+    def fill_items(self, items) -> None:
+        i = 0
+        while True:
+            map_index = i // 144
+            if i == 576 or i == len(items):
+                break
+
+            self._map[map_index].add_item(items[i])
+            i += 1
+
+    def fill_furniture(self, furniture) -> None:
+        i = 0
+        while True:
+            map_index = i // 16
+            if i == 64 or i == len(furniture):
+                break
+
+            self._map[map_index].add_furniture(furniture[i])
+            i += 1
+
+    def fill_terrain(self, terrain) -> None:
+        i = 0
+        while True:
+            map_index = i // 16
+            if i == 64 or i == len(terrain):
+                break
+
+            self._map[map_index].add_terrain(terrain[i])
+            i += 1
+
+    def fill_traps(self, traps) -> None:
+        i = 0
+        while True:
+            map_index = i // 16
+            if i == 64 or i == len(traps):
+                break
+
+            self._map[map_index].add_trap(traps[i])
+            i += 1
+
+    def __str__(self) -> str:
+        retstr = "[\n"
+        for i in range(4):
+            qret = self._map[i].as_string(1)
+            if i != 3:
+                qret += ',\n'
+            retstr += qret
+        retstr += '\n]'
+        return retstr
+
+
+class PathNotExist(Exception):
+    def __init__(self, path: pathlib.Path):
+        super().__init__(f"path does not exist: {path}")
+
+
+class MapsFolderExists(Exception):
+    def __init__(self, path: pathlib.Path):
+        super().__init__(f"maps folder already exists: {path}")
+
+
+class HandlerUnhandledType(Exception):
+    def __init__(self, key):
+        super().__init__(f"key not handled by Handler: {key}")
+
+
+class Handler:
+
+    def __init__(self, path: pathlib.Path, parsed_ids: dict):
+        self.maps_folder = pathlib.Path(args.path)
```

Comment:
It looks quite good overall. Just a few minor improvements visible on the surface: args variable is not passed explicitly to the init here and the path parameter is not used.